### PR TITLE
Tidying Simulator

### PR DESCRIPTION
**Pull request by ghost**
# Description

Cleans up several incorrect implementations and unclear code for simulator. Some notable changes are:

- reverted linear error model back to correct implementation (without random walk)
- selected astropy as desired sun vector function and archived rest
- generalized code to add GNC code to path so that it runs even if your user account isn't eleboeuf

NOTE: This does not solve the issue of adding random walk to the gyroscope. That will be tackled in a separate branch.

### CODE REVIEW

**Comment by hridujain** [comment]
@tomerarnon @NickG-NZ sorry those validation checks gave me some trouble. But all checks now pass and ready for your review on this pull request!

File: Simulator/sensormodels.py [hunk 1]
```diff
@@ -120,16 +120,16 @@ def whitenoise2(mean = 0, cov = 0, dims = None):
...
     
 
-def getTmatrix(scaleF, caSense):
+def getTmatrix(scalefactor, crossaxis_sensitivity):
     """
     Inputs:
-        scaleF: scale factor for a particular sensor
-        caSense: cross-axis sensitivity for particular sensor
+        scalefactor: scale factor for a particular sensor
+        crossaxis_sensitivity: cross-axis sensitivity for particular sensor
```
Comment (kevincrispie) [inline Simulator/sensormodels.py]: I'm not sure it's necessary to have such a long variable name when there is a description here, or whether it's necessary to have a description that's just a restatement of the variable name. Ultimately, it doesn't matter though. I think Tomer put in that original short variable name to keep things cleaner.

Comment (hridujain) [reply Simulator/sensormodels.py]: I personally have no preference. The change was inspired by this discussion: [link](https://github.com/spacecraft-design-lab-2019/groundtruth-simulator/issues/58).

Let me know if you both would like to change it back.

Comment (kevincrispie) [reply Simulator/sensormodels.py]: gotcha, good with me then. Perhaps we should put a note on what cross axis sensitivity and scale factor are, or a link to the datasheet (https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMX160-DS000.pdf), but that is up to you.

File: Simulator/Test/test_kinematics.py [hunk 2]
```diff
@@ -13,9 +14,21 @@
...
+
+	# trivial case
+	q = quat(np.random.rand(4))
+	w = np.zeros(3)
+	np.testing.assert_allclose(calc_q_dot(q,w), np.zeros(4), atol=tol)
 
-	np.testing.assert_allclose(test1, test2, atol=tol)
\ No newline at end of file
+	# against pyquaternion
+	q = quat(np.random.rand(4))
```
Comment (kevincrispie) [inline Simulator/Test/test_kinematics.py]: Is it possible to repeat these random tests more than just once? Not that I'm concerned about it, but bc we have randomness we aren't sure of the actual numbers, but if we ran 50-100 of these then I wouldn't really be worried at all.

Comment (hridujain) [reply Simulator/Test/test_kinematics.py]: Sure, I have no qualms with that. I'll throw it in a for loop.

File: Simulator/Test/test_sun_model.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+def test_sun_position_ECI():
+
+	mjd = 54000
+	dt = julian.from_jd(mjd, fmt='mjd')
+
+	a = sun_model.approx_sun_position_ECI(mjd)
+	b = sun_model.sun_position_ECI(dt)
+	angle_diff = math.acos(np.dot(a,b)/(np.linalg.norm(a)*np.linalg.norm(b)))*180/math.pi
+
+	assert angle_diff < 5
```
Comment (tomerarnon) [inline Simulator/Test/test_sun_model.py]: I just tested this starting on different dates and we exceed 5 degrees. Is there a larger tolerance we should be using, or does this mean the approximation isn't good enough for our purposes? @GathererA

File: Simulator/sensormodels.py (hunk 1 above)
Comment (tomerarnon) [reply Simulator/sensormodels.py]: Good idea to include a link; I'll throw it in there. The datasheet does have cross axis sensitivity in there (searched for it) but doesn't use the phrase scale factor. Do you recall what it calls it?

**Comment by tomerarnon** [comment]
The sun_model test (will) fail(s) since I expanded the test to deal with more dates, and the angle deviation become greater than 5 degrees. Pending solving that little problem (with either a larger tolerance or a tighter estimate), this can be merged.

File: Simulator/Test/test_sun_model.py (hunk 3 above)
Comment (GathererA) [reply Simulator/Test/test_sun_model.py]: weird, shouldn't be that significant, I wish I had time to look into this. What are y'all thinking? Sorry I got to this so late.

Comment (GathererA) [reply Simulator/Test/test_sun_model.py]: Did you compare to JPL ephemerides?

Comment (hridujain) [reply Simulator/Test/test_sun_model.py]: This was resolved by using astropy's built in GCRS to ECI conversion.

**Pull request merged**